packages/react-server-cli/src/parseCliArgs.js

```diff
@@ -119,6 +136,14 @@ const sslize = argv => {
 		}
 	}
 
+	if (argv.https && (argv.httpsKey || argv.httpsCert || argv.httpsCa || argv.httpsPfx || argv.httpsPassphrase)) {
+		throw new Error("If you set https to true, you must not set https-key, https-cert, https-ca, https-pfx, or https-passphrase.");
```

Comment:
If we decide to go all-in with good error handling (probably out-of-scope), maybe a ParseError?